src/assets.rs

```diff
@@ -273,12 +284,26 @@ impl HighlightingAssets {
         self.syntax_set
             .find_syntax_by_extension(file_name.to_str().unwrap_or_default())
             .or_else(|| {
+                let file_path = Path::new(file_name);
                 self.syntax_set.find_syntax_by_extension(
-                    Path::new(file_name)
+                    file_path
                         .extension()
                         .and_then(|x| x.to_str())
                         .unwrap_or_default(),
-                )
+                ).or_else(|| {
+                    let file_str = file_path.to_str().unwrap_or_default();
```

Comment:
This will lead to an empty string if `file_path` contains invalid UTF-8 (which is possible). Instead of directly aborting here, we would still run the full `for` loop below. I think the behavior would still be okay, but it feels a bit strange. And could possibly lead to errors in the future if the code below is changed. I think I would prefer something like:
```rust
if let Some(file_str) = file_path.to_str() {
  // …
}
```

Reply:
SGTM this one too, done.

Thanks a bunch for the guidance and opportunity to learn a bit of Rust :+1: